libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/room/preview/RoomPreviewMapper.kt

```diff
@@ -11,23 +11,28 @@ import io.element.android.libraries.matrix.api.core.RoomAlias
 import io.element.android.libraries.matrix.api.core.RoomId
 import io.element.android.libraries.matrix.api.room.preview.RoomPreview
 import io.element.android.libraries.matrix.impl.room.toRoomType
+import org.matrix.rustcomponents.sdk.JoinRule
+import org.matrix.rustcomponents.sdk.Membership
 import org.matrix.rustcomponents.sdk.RoomPreview as RustRoomPreview
 
 object RoomPreviewMapper {
     fun map(roomPreview: RustRoomPreview): RoomPreview {
-        return RoomPreview(
-            roomId = RoomId(roomPreview.roomId),
-            canonicalAlias = roomPreview.canonicalAlias?.let(::RoomAlias),
-            name = roomPreview.name,
-            topic = roomPreview.topic,
-            avatarUrl = roomPreview.avatarUrl,
-            numberOfJoinedMembers = roomPreview.numJoinedMembers.toLong(),
-            roomType = roomPreview.roomType.toRoomType(),
-            isHistoryWorldReadable = roomPreview.isHistoryWorldReadable,
-            isJoined = roomPreview.isJoined,
-            isInvited = roomPreview.isInvited,
-            isPublic = roomPreview.isPublic,
-            canKnock = roomPreview.canKnock
-        )
+        return roomPreview.use {
+            val info = roomPreview.info()
+            RoomPreview(
+                roomId = RoomId(info.roomId),
+                canonicalAlias = info.canonicalAlias?.let(::RoomAlias),
+                name = info.name,
+                topic = info.topic,
+                avatarUrl = info.avatarUrl,
+                numberOfJoinedMembers = info.numJoinedMembers.toLong(),
+                roomType = info.roomType.toRoomType(),
+                isHistoryWorldReadable = info.isHistoryWorldReadable,
+                isJoined = info.membership == Membership.JOINED,
+                isInvited = info.membership == Membership.INVITED,
+                isPublic = info.joinRule == JoinRule.Public,
+                canKnock = info.joinRule == JoinRule.Knock
```

Comment:
There will probably be a change in the mapping here, but it is out of scope of this PR.